pkg/liquidity-mining/contracts/L2BalancerPseudoMinter.sol

```diff
@@ -0,0 +1,134 @@
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/SingletonAuthentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeMath.sol";
+
+import "./BalancerMinter.sol";
+
+/**
+ * @dev Distributes bridged BAL tokens in child chains, using the same interface as the mainnet Balancer minter.
+ *
+ * This contract is analogous to the mainnet minter: it has the same interface and interacts with (L2) gauges in a
+ * similar manner, keeping track of how many tokens were already distributed to each gauge and user.
+ *
+ * The difference with the mainnet minter is that this contract does not have a way of minting BAL directly: the tokens
+ * are only minted in mainnet, and then bridged to L2s. Then, this contract accumulates the emissions received by
+ * child chain gauges and distributes them to users.
+ *
+ * Every time the tokens are 'minted' (i.e. distributed) from a gauge to a given user, the pseudo minter will query the
+ * child chain gauge for the total amount of tokes that need to be distributed to that user. By keeping track of the
+ * amount that it has already distributed for that gauge / user, the pseudo minter can then transfer the difference
+ * to the user and update the total transferred amount.
+ */
+contract L2BalancerPseudoMinter is BalancerMinter, SingletonAuthentication {
+    event GaugeFactoryAdded(ILiquidityGaugeFactory indexed factory);
+    event GaugeFactoryRemoved(ILiquidityGaugeFactory indexed factory);
+
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    mapping(ILiquidityGaugeFactory => bool) private _validFactories;
+
+    constructor(IVault vault, IERC20 balancerToken)
+        BalancerMinter(balancerToken, "Balancer Pseudo Minter", "1")
+        SingletonAuthentication(vault)
+    {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    /**
+     * @notice Adds a given child chain gauge factory to the allowlist.
+     * @dev This is a permissioned function.
+     * Reverts if the given factory was added beforehand; emits `GaugeFactoryAdded` event upon success.
+     */
+    function addGaugeFactory(ILiquidityGaugeFactory factory) external authenticate {
+        require(!_validFactories[factory], "FACTORY_ALREADY_ADDED");
+        _validFactories[factory] = true;
+        emit GaugeFactoryAdded(factory);
+    }
+
+    /**
+     * @notice Removes a given child chain gauge factory from the allowlist.
+     * @dev This is a permissioned function.
+     * Reverts if the given factory had not been added beforehand; emits `GaugeFactoryRemoved` event upon success.
+     */
+    function removeGaugeFactory(ILiquidityGaugeFactory factory) external authenticate {
+        require(_validFactories[factory], "FACTORY_NOT_ADDED");
+        _validFactories[factory] = false;
+        emit GaugeFactoryRemoved(factory);
+    }
```

Comment:
We might not want this, since removing a factory effectively kills all the gauges created by it (`_updateGauge` would revert).

We already have a `kill` permissioned function for individual gauges, so this would be just a shortcut. I've added it to the initial version because it's a low hanging fruit and for discussion purposes mostly.

Reply:
Would this affect the other reward tokens in any way?

Reply:
No, it should not.
This would make any calls to `mint` (mint for / mint many, etc) revert, but that's only used for BAL emissions.

One detail is that if you call `mintMany`, or `mintManyFor` with a gauge whose factory's been killed, the whole operation would revert, even if there are some valid gauges in the array. On the other hand, calling `mintMany` with killed gauges would not revert, it would just waste some gas.

But you can always choose what gauges to add to the array, so I don't see it as a huge issue anyways.

Reply:
I think this is fine, yes. Thanks!

Reply:
Maybe document that though.